main.go

```diff
@@ -18,17 +17,12 @@ func main() {
 	flag.Parse()
 	//log levels need to be shown correctly in terraform when enabling TF_LOG
 	log.SetFlags(log.Flags() &^ (log.Ldate | log.Ltime))
+	var serveOpts = plugin.ServeOpts{
+		ProviderFunc: ionoscloud.Provider,
+	}
 	if debugMode {
-		err := plugin.Debug(context.Background(), "registry.terraform.io/ionos-cloud/ionoscloud",
-			&plugin.ServeOpts{
-				ProviderFunc: ionoscloud.Provider,
-			})
-		if err != nil {
-			log.Println(err.Error())
-		}
-	} else {
-		plugin.Serve(&plugin.ServeOpts{
-			ProviderFunc: ionoscloud.Provider,
-		})
+		serveOpts.ProviderAddr = "registry.terraform.io/ionos-cloud/ionoscloud"
+		serveOpts.Debug = debugMode
```

Comment:
@cristiGuranIonos again, not something important but for a little bit of clarity you can do:
`serveOpts.Debug = true` but it's not a big deal, do as you feel.

Reply:
will leave it as it is for now